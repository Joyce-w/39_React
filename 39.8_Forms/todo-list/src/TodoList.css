.NewTodoForm-ul-div{
    margin: auto;
    width: fit-content;
    display: flex;
    justify-content: center;
    border: 3px solid grey;
    border-radius: 15px;
    box-shadow: 4px 4px 0 rgba(0, 0, 0, 0.3)
}