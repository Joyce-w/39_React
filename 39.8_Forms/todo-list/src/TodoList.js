import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm"
import Todo from "./Todo"
import { v4 as uuid } from "uuid"
import './TodoList.css'


const TodoList = () => {
    //create initial state value
    const INITIAL_STATE = [
        { 
            id: uuid(),
            task: "Clean fish tank"
        }
    ]

    //create list state form
    const [tasks, setTasks] = useState(INITIAL_STATE);
    console.log(tasks)

    //create update for tasks
    const updateTasks = (task) => {
        setTasks((tasks) => [...tasks, {id:uuid(), task}])
    }

    //remove task function when clicked
    const remove = (e) => {
        let todo = (e.target.parentElement.parentElement.id)
        setTasks(tasks.filter(({ id }) => {
            return id !== todo;
        }))
    }

    return (
        <>
            <NewTodoForm updateTasks={updateTasks} />
            
            <h3>What to do List</h3>
            <div className="NewTodoForm-ul-div">
                <ul className="NewTodoForm-ul">
                {tasks.map(({ id, task }) => {
                    return <Todo key={id} id={id} task={task} remove={ remove }/>
                })}
                </ul>                
            </div>

        </>
    )
}

export default TodoList;