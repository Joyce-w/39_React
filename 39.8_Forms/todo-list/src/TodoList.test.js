import react from "react"
import { render } from '@testing-library/react';
import TodoList from "./TodoList"

//smoketest
test('renders learn react link', () => {
  render(<TodoList />);
});

//snapshot
it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />)
  expect(asFragment()).toMatchSnapshot();
})