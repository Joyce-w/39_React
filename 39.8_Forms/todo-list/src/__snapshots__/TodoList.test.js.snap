// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm-form"
  >
    <label
      for="task"
    >
      New Todo 
    </label>
    <input
      class="NewTodoForm-input"
      id="task"
      name="task"
      placeholder="More things to do"
      type="text"
      value=""
    />
    <button
      class="NewTodoForm-btn"
    >
      Add new task
    </button>
  </form>
  <h3>
    What to do List
  </h3>
  <div
    class="NewTodoForm-ul-div"
  >
    <ul
      class="NewTodoForm-ul"
    >
      <div
        class="Todo"
        data-testid="Todo-test"
        id="6db0f3b7-cdc2-45bd-9d5c-9b05aaa1f6f1"
      >
        <li
          class="Todo-li"
        >
          Clean fish tank
          <button
            class="Todo-btn"
          >
            x
          </button>
        </li>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
